improved reusable code for sound handling

sweeneyapps opened the pull request:
@shosti 

I added reusable code that can be reused for any sound handling in the future.

playSound(soundUrl, soundRepeat) -> return audioPlayer
 - good for reusing for any future notification such as inactive, disconnected, etc.

playSoundOptions( plugin.get('options') ) -> return audioPlayer
  - defined for Sound Notification Options (new work)

stopSound( audioPlayer )
  - to be able to stop specific audioPlayer from repeating.

Also improved playSound testing and for handleWork test, I used window.audioPlayer so I can actually test the sound handling inside handleWork.

sweeneyapps commented:
@shosti if it's possible for you to add a listener on the terminal page to detect if tester actually clicked on the opened workTab (new work) such like   So I can grab info on "worker" inside handleWork.js and stop the notification sound from repeating.

shosti commented on src/chrome/handleWork.js:
```diff
@@ -1,9 +1,10 @@
 
 const handleWork = (store, chrome) => {
   let workTabId = null;
+  window.audioPlayer = null;
```
I'd like to avoid this kind of global variable in general. Is it possible to make it module-local (like `workTabId`)? We shouldn't be sharing state between modules; instead each module should have its own state and react to redux events.

sweeneyapps replied:
I would make it module-local and would have to remove the test I wrote inside handleWork.js test.   Sounds sane?

sweeneyapps replied:
right now, audioPlayer is module-local.

sweeneyapps commented:
I added playSoundOnce() for all notifications.  It will play sound once only user enabled sound on the options page.

sweeneyapps commented:
Summary:
- improved mock test for window.Audio()
- refactored playSound() into reusable code.
- added defined function for sound options for new work.
- added defined function for playing sound once for all another notifications.
- added function to stop sound from repeating, useful for error pages and whenever user is actively working.  (will still need to add event listener to see if user is actively working on the job page)

sweeneyapps commented:
I think it'll be less complicated to just remove stopSound().   stopSound was on my mind if testers choose to set soundRepeat higher.   But you are right, if it's annoying for them, they can lower it.  

So I just commit my changes, and yes..  those function does return audioPlayer because it will make it easier to test them.   The code for playing sound in handleWork.js is back to playSound() without any complications and also for all another notifications, playSoundOnce().
